test/widgets/subscription_list_test.dart

```diff
@@ -190,4 +202,47 @@ void main() {
     check(tester.widget<UnreadCountBadge>(find.byType(UnreadCountBadge)).backgroundColor)
       .equals(swatch);
   });
+
+  testWidgets('muted streams are displayed as faded', (tester) async {
+
+    testOpacityForStreamAndBadge(String streamName, double opacity) {
+      final streamFinder = find.text(streamName);
+      final streamOpacity = tester.widget<Opacity>(
+        find.ancestor(of: streamFinder, matching: find.byType(Opacity))).opacity;
+      final badgeOpacity = tester.widget<Opacity>(find.ancestor(of: find.descendant(
+        of: find.ancestor(of: streamFinder, matching: find.byType(Row)),
+        matching: find.byType(UnreadCountBadge)), matching: find.byType(Opacity))).opacity;
```

Comment:
This is a bit hard to parse visually. It's worth spending a few more lines in order to make the structure more regular:
```suggestion
      final badgeOpacity = tester.widget<Opacity>(find.ancestor(
        of: find.descendant(
          of: find.ancestor(
            of: streamFinder,
            matching: find.byType(Row)),
          matching: find.byType(UnreadCountBadge)),
        matching: find.byType(Opacity)));
```

Or better:
```suggestion
      final badgeOpacity = tester.widget<Opacity>(
        find.ancestor(matching: find.byType(Opacity),
          of: find.descendant(matching: find.byType(UnreadCountBadge),
            of: find.ancestor(matching: find.byType(Row),
              of: streamFinder))));
```

Reply:
(Oh and then `check(badgeOpacity.opacity).equals(opacity)` below.)

Reply:
But also: this ancestor/descendant/ancestor dance feels like it's getting into the implementation details of how this UI is laid out.

How about giving these streams different numbers of unreads? And then this can just be finding the Opacity ancestors of `find.text("2")` and `find.text("1")` respectively.